lib/sequent/core/helpers/date_time_validator.rb

```diff
@@ -0,0 +1,22 @@
+require 'active_model'
+
+module Sequent
+  module Core
+    module Helpers
+      # Validates DateTimes
+      # Automatically included when using a
+      #
+      #   attrs value: DateTime
+      class DateTimeValidator < ActiveModel::EachValidator
+        def validate_each(subject, attribute, value)
+          return if value.is_a?(DateTime)
+          begin
+            DateTime.deserialize_from_json(value)
+          rescue
```

Comment:
I'd like to look at the deserialisers separately. The `"".to_i => 0` is indeed not want we want.

For the rescue: I think in a validator it is okay. For what ever reason we can't parse into a DateTime.